src/attached_probe.cpp

```diff
@@ -1146,7 +1152,7 @@ void AttachedProbe::attach_uprobe(bool safe_mode)
                                         eventname().c_str(),
                                         probe_.path.c_str(),
                                         offset_,
-                                        probe_.pid,
+                                        pid == 0 ? -1 : pid,
```

Comment:
I think that `bpftrace::Probe.pid` is not used anywhere else and we can drop it completely.